### PR TITLE
VIH 7520 Improve the CVP Pipeline

### PR DESCRIPTION
**Pull request by chris-pateman**
These changes are to improve the pipeline and bring more in line with standards.

- higher environments dependent on lower
- testing phases
- sbox has no wait dependency
- prod will only run on master branch
- template breakdown

### CODE REVIEW

File: pipeline/stages/test.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+              scriptLocation: 'inlineScript'
+              inlineScript: |
+                service_name="${{variables.product}}-recordings-${{parameters.env}}-"
+                rgName="${service_name}rg"
+                lbName="${service_name}lb"
+
+                metricName="DipAvailability"
+                resourceType="Microsoft.Network/loadBalancers"
+
+                sleep 2m
```
Comment (lowflying) [inline pipeline/stages/test.yaml:33]: I'm probably missing something obvious but what's with the 2min sleep?

Comment (chris-pateman) [reply pipeline/stages/test.yaml:33]: The Metrics request returns the avg every 1 minute, so leaving it 2 minutes to capture some results before checking it out

File: pipeline/stages/test.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                ruleName="pipelineTestRule"
+
+                echo "Get Source IP"
+                sourceIp="$(dig @resolver4.opendns.com myip.opendns.com +short)"
+                echo "Source IP: $sourceIp"
+
+                echo "Create NSG rule"
+                az network nsg rule create --name $ruleName --nsg-name $nsgName --priority '200' --resource-group $rgName --access 'Allow' --description "Test from the Deployment Pipeline"  --destination-port-ranges '443' --direction 'Inbound' --protocol 'Tcp' --source-address-prefixes $sourceIp  --source-port-ranges '*'
+
+                sleep 20s
```
Comment (lowflying) [inline pipeline/stages/test.yaml:129]: and conversly, I've had issues with security group rules taking up to a minute to apply. Reckon this needs a check added here? Or maybe just a comment saying rerun if this fails? If you check some the azure-devops-templates/jobs/azWebAppdeloyJob the firewall adding/removal there is a bit OTT

File: pipeline/stages/test.yaml (hunk 1 above)
Comment (lowflying) [reply pipeline/stages/test.yaml:33]: ahhhh, so yes, I was missing something obvious :D

File: pipeline/stages/test.yaml (hunk 2 above)
Comment (chris-pateman) [reply pipeline/stages/test.yaml]: Loop add round check for nsg rule to be added

**Review by vialde — APPROVED** [review]

**Pull request merged**